Update label processing4

wakco opened the pull request:
New Team ID

All questions must be filled out or your Pull Request will be closed for lack of information. The first three questions should be answered `Yes` before submitting the pull request.
---
**Have you confirmed this pull request is not a duplicate?**
Yes

**Is this pull request creating or modifying a label in the fragments/labels folder, and not Installomator.sh itself?**
Yes

**Did you use [our editorconfig file](https://github.com/Installomator/Installomator/wiki/Contributing-to-Installomator)?**
No, only the Team Id has been updated.

**Additional context** Add any other context about the label or fix here.
#2209 references both processing3 and processing4, but when I checked, only processing4 has the wrong Team ID

**Installomator log** At the bottom of this pull request, provide a log of a label run by running Installomator in Terminal and saving the output. `DEBUG=1` can be enabled but **do not enable [Debug logging level](https://github.com/Installomator/Installomator/wiki/Configuration-and-Variables#logging-level) and please format the log [using a code block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#fenced-code-blocks)!**

Please identify any issues fixed by your pull request by including the issue number. (Example: "Fixes #XXXX")
Fixes #2209 

I've skipped the log, as it's just an `expectedTeamID` fix.

wakco committed:
Update processing4.sh

New Team ID

hwlans commented:
Hi @wakco , may I suggest two improvements to this PR:

switch from zip to `type="dmg"`
in `archiveName`, replace .zip with .dmg

Processing4 [switched to dmg](https://github.com/processing/processing4/releases/tag/processing-1300-4.4.0) as of version 4.4.x. This would fix #2475

uurazzle reviewed (changes requested):
Hi @wakco,

Thanks for the PR. 

### Resolve debug execution failure

I ran the submitted label through the Installomator debug execution check and it exited non-zero:
- `processing4`: exit 8

For example...

```
--- Debug execution log: processing4 (exit 8) ---
         2026-05-06 14:31:36 : INFO  : processing4 : setting variable from argument DEBUG=1
         2026-05-06 14:31:36 : INFO  : processing4 : setting variable from argument LOGGING=DEBUG
         2026-05-06 14:31:36 : INFO  : processing4 : setting variable from argument NOTIFY=silent
         2026-05-06 14:31:36 : INFO  : processing4 : setting variable from argument BLOCKING_PROCESS_ACTION=ignore
         2026-05-06 14:31:36 : INFO  : processing4 : Total items in argumentsArray: 4
         2026-05-06 14:31:36 : INFO  : processing4 : argumentsArray: DEBUG=1 LOGGING=DEBUG NOTIFY=silent BLOCKING_PROCESS_ACTION=ignore
         2026-05-06 14:31:36 : REQ   : processing4 : ################## Start Installomator v. 10.9beta, date 2026-05-06
         2026-05-06 14:31:36 : INFO  : processing4 : ################## Version: 10.9beta
         2026-05-06 14:31:36 : INFO  : processing4 : ################## Date: 2026-05-06
         2026-05-06 14:31:36 : INFO  : processing4 : ################## processing4
         2026-05-06 14:31:36 : DEBUG : processing4 : DEBUG mode 1 enabled.
         2026-05-06 14:31:39 : INFO  : processing4 : Reading arguments again: DEBUG=1 LOGGING=DEBUG NOTIFY=silent BLOCKING_PROCESS_ACTION=ignore
         2026-05-06 14:31:39 : DEBUG : processing4 : argument: DEBUG=1
         2026-05-06 14:31:39 : DEBUG : processing4 : argument: LOGGING=DEBUG
         2026-05-06 14:31:39 : DEBUG : processing4 : argument: NOTIFY=silent
         2026-05-06 14:31:39 : DEBUG : processing4 : argument: BLOCKING_PROCESS_ACTION=ignore
         2026-05-06 14:31:39 : DEBUG : processing4 : name=Processing
         2026-05-06 14:31:39 : DEBUG : processing4 : appName=
         2026-05-06 14:31:39 : DEBUG : processing4 : type=zip
         2026-05-06 14:31:39 : DEBUG : processing4 : archiveName=Processing-[0-9.]*-macOS-aarch64.zip
         2026-05-06 14:31:39 : DEBUG : processing4 : downloadURL=https://github.com
         2026-05-06 14:31:39 : DEBUG : processing4 : curlOptions=
         2026-05-06 14:31:39 : DEBUG : processing4 : appNewVersion=13134.5.2
         2026-05-06 14:31:39 : DEBUG : processing4 : appCustomVersion function: Defined. 
         2026-05-06 14:31:39 : DEBUG : processing4 : versionKey=CFBundleShortVersionString
         2026-05-06 14:31:39 : DEBUG : processing4 : packageID=
         2026-05-06 14:31:39 : DEBUG : processing4 : pkgName=
         2026-05-06 14:31:39 : DEBUG : processing4 : choiceChangesXML=
         2026-05-06 14:31:39 : DEBUG : processing4 : expectedTeamID=6297K33652
         2026-05-06 14:31:39 : DEBUG : processing4 : blockingProcesses=
         2026-05-06 14:31:39 : DEBUG : processing4 : installerTool=
         2026-05-06 14:31:39 : DEBUG : processing4 : CLIInstaller=
         2026-05-06 14:31:39 : DEBUG : processing4 : CLIArguments=
         2026-05-06 14:31:39 : DEBUG : processing4 : updateTool=
         2026-05-06 14:31:39 : DEBUG : processing4 : updateToolArguments=
         2026-05-06 14:31:39 : DEBUG : processing4 : updateToolRunAsCurrentUser=
         2026-05-06 14:31:39 : INFO  : processing4 : BLOCKING_PROCESS_ACTION=ignore
         2026-05-06 14:31:39 : INFO  : processing4 : NOTIFY=silent
         2026-05-06 14:31:39 : INFO  : processing4 : LOGGING=DEBUG
         2026-05-06 14:31:40 : INFO  : processing4 : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
         2026-05-06 14:31:40 : INFO  : processing4 : Label type: zip
         2026-05-06 14:31:40 : INFO  : processing4 : archiveName: Processing-[0-9.]*-macOS-aarch64.zip
         2026-05-06 14:31:40 : INFO  : processing4 : no blocking processes defined, using Processing as default
         2026-05-06 14:31:40 : DEBUG : processing4 : Changing directory to /private/var/folders/r7/gm7k1lvx1178n0l2rcf61l940000gp/T/installomator-lint.4c7MXs/pr-verify-2309/build
         2026-05-06 14:31:40.097 defaults[47111:37353499] 
         The domain/default pair of (/Applications/Processing.app/Contents/Info.plist, CFBundleVersion) does not exist
         2026-05-06 14:31:40.108 defaults[47112:37353504] 
         The domain/default pair of (/Applications/Processing.app/Contents/Info.plist, CFBundleShortVersionString) does not exist
         2026-05-06 14:31:40 : INFO  : processing4 : Custom App Version detection is used, found 
         2026-05-06 14:31:40 : INFO  : processing4 : appversion: 
         2026-05-06 14:31:40 : INFO  : processing4 : Latest version of Processing is 13134.5.2
         2026-05-06 14:31:40 : REQ   : processing4 : Downloading https://github.com to Processing-[0-9.]*-macOS-aarch64.zip
         2026-05-06 14:31:40 : DEBUG : processing4 : No Dialog connection, just download
         2026-05-06 14:31:41 : INFO  : processing4 : Downloaded Processing-[0-9.]*-macOS-aarch64.zip – Type is  HTML document text, Unicode text, UTF-8 text, with very long lines (30799) – SHA is da85dbc716f475fb4d8b7b476bb222f8befbf22c – Size is 556 kB
         2026-05-06 14:31:41 : INFO  : processing4 : ignoring blocking processes
         2026-05-06 14:31:41 : REQ   : processing4 : Installing Processing
         2026-05-06 14:31:41 : INFO  : processing4 : Unzipping Processing-[0-9.]*-macOS-aarch64.zip
         ditto: Couldn't read PKZip signature
         2026-05-06 14:31:41 : DEBUG : processing4 : DEBUG mode 1, not reopening anything
         2026-05-06 14:31:41 : ERROR : processing4 : ERROR: could not find: /private/var/folders/r7/gm7k1lvx1178n0l2rcf61l940000gp/T/installomator-lint.4c7MXs/pr-verify-2309/build/Processing.app
         2026-05-06 14:31:41 : REQ   : processing4 : ################## End Installomator, exit code 8 
         
--- end debug execution log ---

```

Please run the label locally from Terminal and paste the full Installomator output in this PR wrapped in a fenced code block. `DEBUG=1` is fine if helpful, but please do not set `LOGGING=DEBUG`.

```bash
./assemble.sh <label> DEBUG=1
```

```code
paste full output here
```

Once we have the full log, we can tell whether this is a label issue, vendor download issue, or an expected exit code. Thanks!

wakco committed:
Update processing4.sh

A lot has changed in the last year, it's now an app in a dmg, and they no longer include the build number in the version within the app while still including it in the github version, so no longer needed appCustomVersion, but for a version match, could no longer use the standard github version response, so instead since it's in the download url in a way that is easier to extract than the github version, we'll use that instead.

uurazzle reviewed (approved):
Hi @wakco,

Thanks for the updates on this Installomator label. I am approving this PR,

gilburns reviewed (approved):
Thank you for your contribution to the project!
